sklearn/linear_model/ridge.py

```diff
@@ -1324,17 +1326,21 @@ def _solve_eigen_covariance_intercept(
         return (1 - hat_diag) / alpha, (y - y_hat) / alpha
 
     def _solve_eigen_covariance(
-            self, alpha, y, sqrt_sw, X_mean, s, V, X):
-        """Compute dual coefficients and diagonal of (Identity - Hat_matrix)
+            self, alpha, y, sqrt_sw, X_mean, eigvals, Q, X):
+        """Compute dual coefficients and diagonal of (Identity - Hat_matrix),
+        where Hat_matrix = X(X^TX + alpha*I)^(-1)X^T
 
         Used when we have a decomposition of X^T.X
-        (n_features < n_samples and X is sparse).
+        (n_samples > n_features and X is sparse).
+
+        Letting X^T.X = QVQ^T, the Hat_matrix can be written as:
+        QV(V + alpha*I)^(-1)VQ
```

Comment:
maybe X = USV^T, and the influence or hat matrix is
XV (1 / (S^2 + alpha I)) V^TX^T

Reply:
(Copy from another comment)

My understanding is, in `_eigen_decompose_covariance` we are doing an eigenvalue decomposition of `X^TX` and `_solve_eigen_covariance` uses this to get 

```
X(X^TX + alpha*I)^(-1)X^T
```

Letting `X^TX=QLQ^T`, we write this as 

```
XQ^T(L + alpha*I)^(-1)QX^T
```

I may be missing something. Where are we doing SVD directly in this procedure?

Reply:
I can see this can be put in the context of using the right singular matrix, but we only actually call `svd` in `_svd_decompose_design_matrix` (not in `_eigen_decompose_covariance`).

Reply:
still `_eigen_decompose_covariance` uses `V`, the eigenvectors of `X^TX`, rather than `Q`, which are the eigenvectors of `XX^T`. Either way in the docstring this line:
```
         QV(V + alpha*I)^(-1)VQ 
```
needs to be corrected, replaced by 
```
XQ(L + alpha*I)^(-1)Q^TX^T
```
if we keep the notation you suggest, or
```
XV(S**2 + alpha*I)^(-1)V^TX^T
```
to avoid using `Q` for both the left and right singular vectors of `X`

Reply:
Okay I see what you mean with the Q. Using V looks good for me. Using `S**2` is still al little suspect as mention in the other comment.

Reply:
> Okay I see what you mean with the Q. Using V looks good for me. Using `S**2` is still al little suspect as mention in the other comment.

cool! sorry if I was unclear. If you dislike `S**2` I think any other letter is fine, outside of `Q`, `V`, `U`, `X`, `S` and `L`